src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.producer.implementations.BulldozerProducer;
+import core.mate.academy.producer.implementations.ExcavatorProducer;
+import core.mate.academy.producer.implementations.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type == Truck.class) {
+            machines = new TruckProducer().get();
+            return machines;
```

Comment:
```suggestion
            return new TruckProducer().get();
```
the same for other if blocks